core/src/main/web/app/mainapp/components/pluginmanager/pluginmanager-directive.js

```diff
@@ -145,5 +145,23 @@
       }
     };
 
+    $rootScope.$on(GLOBALS.EVENTS.LANGUAGE_MANAGER_SHOW_SPINNER, function(event, data) {
+      $scope.showSpinner = true;
+      $scope.showMessage = true;
+      $scope.loadingMessage = 'Restarting ' + data.pluginName + '...';
+    });
+
+    $rootScope.$on(GLOBALS.EVENTS.LANGUAGE_MANAGER_HIDE_SPINNER, function(event, data) {
+      if (data.error) {
+        $scope.loadingMessage += 'failed';
+      } else {
+        $scope.loadingMessage += 'done';
+      }
+      $scope.showSpinner = false;
+      bkUtils.timeout(function() {
+        $scope.showMessage = false;
+      }, 3000);
```

Comment:
I just bumped into this, I guess I either didn't notice it, or didn't understand it before.

We seem to be guessing here that the language reset operation will take ~3s, right? Isn't there a mechanism in place that tells us when the environment is ready? Like when we first start the language, we poll to determine if backend is ready.